### ISSUE TITLE
termit initialization fails to include colormaps and utils

### ISSUE THREAD

**Issue by friartek** [nonstop/termit#105]
I built termit on a Funtoo system.  This took a little work trying to meet all the dependencies, but the build finally worked.  My home directory had my own _rc.lua_ and _colormaps.lua_ files from using termit in the past on a Debian Wheezy system.  Using **Applications >> System >> Termit** to start the termit I didn't see what I expected.  The window was very small do to the very small font that it used.  Termit was using some defaults, I assume internal to the code.

Back to basics.  Moved my personal files so the termit would default to the files in /etc.  Here too, I got the same results.  Starting termit from a terminal I was able to see the error messages.  Troubleshooting revealed that the require statements in _rc.lua_ failed when the files colormaps or utils  were read.

After searching the web I found that **module** was deprecated in lua 5.2(maybe even in later versions of 5.1).  BTW, my build of termit uses lua version 5.3.  So the statements `module('termit.colormaps')` and `module('termit.utils')` in their respective files failed with the message `attempt to call a nil value (global 'module')`.

As I am still learning lua, it took some searching the web to come up with a solution that would work under my environment.  I'm not sure it would work under all earlier versions of lua, and to be honest I'm guessing there is most likely a better way.  I will have to leave it to someone else to come up with a way to make it work under different versions of lua.

_/etc/xdg/termit/rc.lua_:

Added to the top of the file:
  `package.path = package.path .. ';/etc/xdg/termit/?.lua'`
Replaced `require("termit.colormaps")` with `require "colormaps"`
Replaced `require("termit.utils")` with `require "utils"`

_/etc/xdg/termit/colormaps.lua_:

Added to the top of the file:  `local colormaps = {}`
Added to the bottom of the file:  `return colormaps:`
Replaced `module('termit.colormaps')` with `local colormaps = {}`

_/etc/xdg/termit/utils.lua_:

Added to the top of the file:  `local utils = {}`
Added to the bottom of the file:  `return utils:`
Replaced `module('termit.colormaps')` with `local utils = {}`

This is only a **work around**!  As I said I'm not a lua programmer.  These work arounds cause other issues.  For example the _rc.lua.example_ file would also need to be changed as in _/etc/xdg/termit/rc.lua_ above.  But also the statements like `termit.colormaps` or `termit.utils` would have to be changed `colormaps` and `utils` respectively.

I hope someone can come up with a better fix for this issue.

Sincerely,

Jim "Friartek"

System information:

Distribution:  funtoo
kernel-name:  Linux
nodename:  mysystem
kernel-release:  4.5.2-1
kernel-version:  #1 SMP Fri May 6 06:19:05 2016
machine:  x86_64
processor:  Intel(R) Core(TM) i7-4710MQ CPU @ 2.50GHz
hardware-platform:  GenuineIntel
operating-system:  GNU/Linux

termit:  3.0.0
vte:  0.44.2
lua; 5.3.3
zsh: 5.3.1
xfc4:  4.12

**Comment by nonstop** [comment]
It seems termit should require Lua-5.2 only.

**Comment by friartek** [comment]
Hello nonstop,

First, thank you for all your work on termit.  I missed it when moving to Funtoo, but I'm glad I was able to get it up and running.

As I said, **module** has been deprecated since 5.2-beta as far as I can tell.  It would appear that there was/is a grace period as I found a statement on the Lua users mailing list stating:
  (a) **module** is deprecated
  (b) even though, **module** is still defined and enabled by default...
I believe this was mainly do to the impact it would have on code already written.

Eventually this will have to be addressed.  I would guess that in 5.3 that **module** is no longer defined or at the least not enabled by default.

I could be wrong, but it seems there are only four(4) Lua files that ship with termit:
  doc/rc.lua.example
  etc/termit/colormaps.lua
  etc/termit/rc.lua
  etc/termit/utils.lua
With a total of 194 lines(wc -l) of code, and only a few are related to this issue, it would seem that this should be resolvable without a lot of work for someone who knows Lua.  I am a Lua novice, and I was able to come up with a work around.  Not elegant, but allowed me to configure termit. My personal init file(rc.lua) is not all that big and I am still making changes to it, but so far no snags. If nothing else this has been a good learning experience in Lua coding.

As far as termit's functionality and usability, It doesn't seem to be affected by Lua-5.3. If starting termit from a terminal I have seen a few vte and gtk warnings, but termit has not crashed or shown any other issues as far as I can tell.  Other then this one issue I have been quite happy with the way termit has functioned.

I have no plans, at this time, of downgrading to Lua-5.2.  I do plan on researching what is needed to remove **module** and replace it with code that will not need the additional _package.path_ entries.  If anyone has any suggestions, I would be open to any and all input.  As I am still learning Lua it may take me a while to complete.

I don' know if you want to keep this issue open, or at least put it on a **todo** list. I do believe it is something that needs to me addressed.

Sincerely,

Jim "Friartek"

**Comment by nonstop** [comment]
#104 concernes lua-5.3 also
I try to investigate the problem as a whole
lua-5.3 requires special moves for sure

**Comment by friartek** [comment]
Thanks for the feedback.  After some more research I started a second workaround that no longer requires adding additional paths to `package.path`,  So far the changes I have made are all Lua related.  Haven't been able to get the **Encodings** popup to work, but not a show stopper.  I will keep working on that.  As soon as I get the **Encodings** popup to work, I will check out the other functions in utils.lua and let you know what else may need code changes.  I will get back to you when I have more information.  Hopefully the changes I make will be of some help.

So far this build of termit has been very usable.
